modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,106 @@
+* On the target cluster, the `ImageBasedUpgrade` CR specifies the seed container image for the upgrade of the target cluster and the backup configurations for your workloads.
+
+.Example SeedGenerator CR
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: SeedGenerator
+metadata:
+  name: seedimage
+spec:
+  seedImage: <seed_container_image>
+----
+
+.Example ImageBasedUpgrade CR
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  generation: 1
+  name: upgrade
+spec:
+  stage: Idle
+status:
+  conditions:
+  - lastTransitionTime: "2024-01-19T05:42:58Z"
+    message: Idle
+    observedGeneration: 1
+    reason: Idle
+    status: "True"
+    type: Idle
+  observedGeneration: 1
+----
+
+After generating the seed image on the seed cluster, you can move through the stages on the target cluster by setting the `spec.stage` field to the following values in the `ImageBasedUpgrade` CR:
+
+* `Idle`
+* `Prep`
+* `Upgrade`
+* `Rollback` (Optional)
+* `Idle`
+
+[discrete]
+== Idle stage
+
+This is the default stage when the {lcao} is installed. There is no ongoing upgrade and the cluster is ready to move to the `Prep` stage.
+
+After a successful upgrade or a rollback, you commit to the change by manually moving the stage to `Idle`.
+Changing to this stage ensures that the {lcao} cleans up resources, so the cluster is ready for upgrades again.
+
+[discrete]
+== Prep stage
+
+You can complete this stage before a scheduled maintenance window.
+
+In the `Prep` stage, the {lcao} ensures that the target cluster is ready to proceed to the `Upgrade` stage by checking if it meets certain conditions.
+The Operator uses the `ostree` versioning system and the OADP Operator to create a backup, which allow complete cluster reconfiguration after upgrade or rollback.
+
+At the `Prep` stage, the {lcao} pulls the seed image and additional container images specified in the seed image to the target cluster.
+
+You can stop the upgrade process.
+If you stop, the Operator performs cleanup operations and sets the upgrade stage back to `Idle`.
+
+[discrete]
+== Upgrade stage
+
+Before the {lcao} starts the upgrade process, you can create an optional backup of your cluster resources with the OADP Operator.
+The OADP Operator creates a backup of the user-specified resources on a compatible Object storage solution, which are defined in the `Backup` CR.
+After the target cluster reboots with the new platform version, the Operator applies the cluster and application configurations defined in the `Restore` CR.
+
+The Operator also regenerates the cluster certificate.
+This ensures that each {sno} cluster upgraded with the same seed image has unique and valid cryptographic objects.
+
+[discrete]
+== (Optional) Rollback stage
+
+The rollback stage can be initiated manually or automatically.
```

Comment:
Should elaborate on the automatic rollback. This is initiated if the LCA has detected the upgrade has failed or is taking longer than the specified limit.